packages/NodeTypeResolver/tests/LazyTypeAttributeTest.php

```diff
@@ -0,0 +1,46 @@
+<?php declare(strict_types=1);
+
+namespace Rector\NodeTypeResolver\Tests;
+
+use PhpParser\Node\Expr\Variable;
+use Rector\Node\Attribute;
+
+final class LazyTypeAttributeTest extends AbstractNodeTypeResolverTest
+{
+    /**
+     * @var bool
+     */
+    private $isCalled = false;
+
+    /**
+     * @var Variable
+     */
+    private $variableNode;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $nodes = $this->getNodesForFileOfType(__DIR__ . '/LazyTypeAttributeSource/MethodCall.php.inc', Variable::class);
+        $this->assertCount(3, $nodes);
+
+        $this->variableNode = $nodes[2];
+    }
+
+    public function testLazyPrototype(): void
+    {
+        $this->variableNode->setAttribute('someKey', function () {
```

Comment:
We need to search a CS for return types in anonymous functions :thinking:

Reply:
@TomasVotruba [Is this](https://github.com/slevomat/coding-standard/blob/master/SlevomatCodingStandard/Sniffs/TypeHints/TypeHintDeclarationSniff.php#L26) what we are looking for? 

I don't know how PHP_CodeSniffer works and how we could include it, but if you can guide me :man_teacher: